popularity-calculation-script/src/popularity_scraper.py

```diff
@@ -1,111 +1,99 @@
 import json
 import re
-from urllib.request import urlopen
 
 import pandas as pd
+import requests
 
 
-class PopularityScraper:
-    def __init__(self, url: str, filename: str) -> None:
+class PopularityCalculator:
+    def __init__(self, url: str):
         self.url = url
-        self.filename = filename
         self.df = pd.DataFrame()
 
-    def download_data_from_kiwix_url(self, label="download.kiwix.org") -> dict:
-        """Get the relevent dictionery from the list of dictioneries from the data
-
-        Args:
-            label (str, optional): Label key of dictionary. Defaults to "download.kiwix.org".
-
-        Returns:
-            dict: the inner dictionary with the relevent label
-        """
-        response = urlopen(self.url)
-        data_json = json.loads(response.read())
+    def __download_data_from_url(self):
+        """Download data from url and append it to the Pandas DataFrame"""
+        response = requests.get(self.url)
+        data_json = json.loads(response.content)
         for i in data_json:
-            i["label"] == label
-            subtable = i["subtable"]
-            self.df = pd.DataFrame(subtable)
-            break
+            # adding the data (dicts) to the dataframe
+            self.df = self.df.append(pd.DataFrame(i["subtable"]), ignore_index=True)
 
-    def remove_nonzim_data_from_df(self):
+    def __remove_nonzim_data_from_df(self):
         # clean the data; remove rows that don't have "zim" substring in the label column
         ## refer: https://stackoverflow.com/questions/17097643/search-for-does-not-contain-on-a-dataframe-in-pandas
         self.df = self.df.drop(
             self.df[~self.df.label.str.endswith(".zim.torrent")].index
         )
 
-    def add_zim_name_column(self):
+    def __add_zim_name_column(self):
         for index, row in self.df.iterrows():
+            # adding a new column named "zim", and setting value in it for each row
             self.df.at[index, "zim"] = self.extract_zim_name_from_label(row["label"])
 
-    def extract_zim_name_from_label(self, label: str) -> str:
-        """Get the zim name from the label without the slash prefixes and date
-
-        Args:
-            label (str): Label of the row
-
-        Returns:
-            str: zim name
+    @staticmethod
+    def extract_zim_name_from_label(label: str) -> str:
+        """Get the zim name from the label, i.e. without the slash prefixes and date.
+        Expected output: "wikipedia_ar_all_maxi"
```

Comment:
There should be a blank line after line 1 in a docstring

Reply:
Done